src/file-utils.ts

```diff
@@ -42,12 +43,32 @@ export function storeFileExtensionConfig(workspaceDir: string, fileExtensions: s
     }
 }
 
+/**
+ * @param scriptPath full file path to script that is exported
+ * @param fileExtensions supported file extensions of the DataPlugin
+ * @param exportPath full path to
+ * @param embedScript embed the script inside uri file
+ */
 export async function writeUriFile(
     scriptPath: string,
     fileExtensions: string,
-    exportPath: string
+    exportPath: string,
+    embedScript = false
 ): Promise<void> {
-    const dirName = path.basename(path.dirname(scriptPath));
-    const uriTemplate = new UriTemplate(`${dirName}`, scriptPath, fileExtensions);
+    const pluginName = path.basename(path.dirname(scriptPath));
+
+    let uriTemplate: UriTemplate;
+    if (embedScript) {
+        const fileContent = fs.readFileSync(scriptPath, { encoding: 'utf8' });

```

Comment:
What happens if the file isn't `utf8` encoded?

Reply:
If the python code is utf16, it won't work. However, python code shouldn't be utf16. I believe this is very uncommon and not recommended.